### PR TITLE
Add api_key validation in aiohttp and fastapi examples

### PR DESCRIPTION
**Pull request by lilida**
Added the api_key validation in both aiohttp and fastapi. Tested in bot.

### CODE REVIEW

File: fastapi_poe/src/fastapi_poe/base.py [hunk 1]
```diff
@@ -48,6 +62,19 @@ def exception_handler(request: Request, ex: HTTPException):
     logger.error(ex)
 
 
+http_bearer = HTTPBearer()
+
+
+def auth_user(authorization: str = Depends(http_bearer)) -> None:
+    logger.info(f"auth is called key is {auth_key} and credition is {authorization}")
```
Comment (JelleZijlstra) [inline fastapi_poe/src/fastapi_poe/base.py]: Probably shouldn't log the auth key for security

File: fastapi_poe/src/fastapi_poe/base.py [hunk 2]
```diff
@@ -23,23 +25,35 @@
...
+
     async def dispatch(self, request: Request, call_next):
         logger.info(f"Request: {request.method} {request.url}")
         try:
+            # Per https://github.com/tiangolo/fastapi/issues/394#issuecomment-927272627
+            # to avoid blocking.
+            await self.set_body(request)
             body = await request.json()
-            logger.info(f"Request body: {json.dumps(body)}")
+            logger.debug(f"Headers: {request.headers}")
```
Comment (JelleZijlstra) [inline fastapi_poe/src/fastapi_poe/base.py]: Let's not log the headers since the API key is sensitive

File: aiohttp_poe/src/aiohttp_poe/base.py [hunk 3]
```diff
@@ -38,6 +39,32 @@ async def prepare(self, request: web.Request):
...
+    if auth_header:
+        scheme, _, token = auth_header.partition(" ")
+        if scheme.lower() == "bearer":
+            is_authenticated = await authenticate(request, token)
+            if is_authenticated:
+                return await handler(request)
+    return web.HTTPUnauthorized(headers={"WWW-Authenticate": "Bearer"})
+
+
+def auth_user(web_request: web.Request) -> None:
```
Comment (JelleZijlstra) [inline aiohttp_poe/src/aiohttp_poe/base.py]: This function is unused

**Review by JelleZijlstra — COMMENTED** [review]
Thanks!

File: fastapi_poe/src/fastapi_poe/base.py (hunk 1 above)
Comment (lilida) [reply fastapi_poe/src/fastapi_poe/base.py]: Yeah. Definitely. Should catch this earlier.

**Pull request merged**